README.adoc

```diff
@@ -4336,6 +4336,26 @@ Use non-capturing groups when you don't use the captured result.
 /(?:first|second)/
 ----
 
+=== No mix named and numbered captures [[no-mix-named-and-numbered-captures]]
+
+Do not mix named captures and numbered captures in a Regexp literal.
+Because numbered capture is ignored if they're mixed.
+
+[source,ruby]
+----
+# bad
+/(?<foo>FOO)(BAR)/
+
+# good
+/(?<foo>FOO)(?<bar>BAR)/
+
+# good
```

Comment:
I'd add some comments explaining examples like this one, as I'm pretty sure many people are not familiar with non-capturing groups.